### PR TITLE
drop support for Python 3.7

### PR DESCRIPTION
**Pull request by pmeier**
According to https://github.com/pytorch/pytorch/issues/80513#issuecomment-1396166181, core will drop support for 3.7 with the upcoming `torch==2.0.0` release. We should do the same.

cc @seemethere

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -58,7 +58,6 @@ def write_version_file():
     pytorch_dep += "==" + os.getenv("PYTORCH_VERSION")
 
 requirements = [
-    "typing_extensions",
```
Comment (pmeier) [inline setup.py:61]: We only ever used this for `Literal`, which is available through `typing` in the standard library for Python >= 3.8.

**Comment by pmeier** [comment]
We have some failing 3.7 jobs that we have no influence over. They are generated [here](https://github.com/pytorch/test-infra/blob/a58debdc3837ac78ec4186ea57378391b4b1efa6/tools/scripts/generate_binary_build_matrix.py#L23) from https://github.com/pytorch/vision/blob/93df9a50885d0345e31bba691576c83d5cee7737/.github/workflows/build-wheels-linux.yml#L12. Thus, this PR is blocked by pytorch/test-infra#1416.

**Comment by pmeier** [comment]
pytorch/test-infra#1416 was closed in favor of pytorch/test-infra#1475. IIUC, this was done since PyTorch core tests stable binaries as well: https://github.com/pytorch/test-infra/pull/1416#issuecomment-1402786799. I'm fairly certain we don't so we should be good to remove everywhere. cc @atalman and @malfet to be sure.

**Comment by pmeier** [comment]
CI failures seem unrelated. I'll deal with them today.

**Review by malfet — APPROVED** [review]
Looks good to me, but I wonder if we it'll be at all possible possible use more templating to make this change smaller in the future

**Review by NicolasHug — APPROVED** [review]
Thanks Philip!

**Pull request merged**